src/Servers/Kestrel/Core/src/Internal/HttpConnection.cs

```diff
@@ -269,11 +275,11 @@ public void OnTimeout(TimeoutReason reason)
                 break;
             case TimeoutReason.WriteDataRate:
                 Log.ResponseMinimumDataRateNotSatisfied(_context.ConnectionId, _http1Connection?.TraceIdentifier);
-                Abort(new ConnectionAbortedException(CoreStrings.ConnectionTimedBecauseResponseMininumDataRateNotSatisfied));
+                Abort(new ConnectionAbortedException(CoreStrings.ConnectionTimedBecauseResponseMininumDataRateNotSatisfied), ConnectionEndReason.MinResponseDataRate);
                 break;
             case TimeoutReason.RequestBodyDrain:
             case TimeoutReason.TimeoutFeature:
-                Abort(new ConnectionAbortedException(CoreStrings.ConnectionTimedOutByServer));
+                Abort(new ConnectionAbortedException(CoreStrings.ConnectionTimedOutByServer), ConnectionEndReason.ServerTimeout);
```

Comment:
I left them the same because there is one log message for both scenarios. We can add more in the future, so I think it's best to leave them grouped together and then split them up if there is feedback.